app/src/main/java/org/mozilla/reference/browser/BrowserApplication.kt

```diff
@@ -73,7 +74,23 @@ private fun setupGlean(context: Context) {
     Glean.initialize(context, Configuration(httpClient = lazy { context.components.core.client }))
     Glean.setUploadEnabled(BuildConfig.TELEMETRY_ENABLED && Settings.isTelemetryEnabled(context))
     GleanFactProcessor().register()
-    Experiments.initialize(context, ExperimentsConfiguration(httpClient = lazy { context.components.core.client }))
+    Experiments.initialize(
+        context,
+        ExperimentsConfiguration(httpClient = lazy { context.components.core.client })
+    )
+
+    // Recording the experiment ID in Glean through the use of the `withExperiment` function should
+    // allow us to validate the automatically recorded enrollment in the experiment, as well as the
+    // functionality of doing something within the client app based on the specific branch. It
+    // should be noted that the first time that the client is enrolled in the experiment, the
+    // following code will not be executed as this only gets called on startup, so it will be on
+    // the next application launch following enrollment that the code below will be executed and the
+    // metric recorded.
+    Experiments.withExperiment("reference-browser-test") {
+        // If we match the 'reference-browser-test' experiment, 'it' will be the branch name that
+        // we were enrolled in.
+        ExperimentsMetrics.activeExperiment.set(it)
```

Comment:
Would it be possible to use a more readable name for the parameter instead of the implicit `it`?

```kotlin
Experiments.withExperiment("reference-browser-test") { branchName ->
    ExperimentsMetrics.activeExperiment.set(branchName)
}
```

Reply:
Certainly!